qml/components/CustomSpinBox.qml

```diff
@@ -21,26 +21,33 @@
 ********************************************/
 
 import QtQuick 2.12
-import QtQuick.Controls 2.4
+import QtQuick.Controls 2.5
 import "../components"
 import "../components/helpers"
 import "../style"
+import "../utils"
 
 SpinBox {
     id: spinBox
     property string prefix
     property string suffix
 
-    property int maxCharCount: prefix.length + suffix.length + charCount(to)
+    readonly property real averageWidth: UiUtils.averageTextWidth(d.fontMetrics, d.maxCharCount) + 2 * spacing + up.implicitIndicatorWidth
```

Comment:
Looks fishy, what is the reason behind it?

Reply:
Not that fishy ;) It just provides a width that in most cases (i.e. avarage width) can fit all the characters.